### ISSUE TITLE
Ability to not overwrite CLI deployments

### ISSUE THREAD

**Issue by sti0** [PrefectHQ/prefect#6644]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently if a deployment gets executed multiple times it overwrites settings done in the UI. This cause empty or maybe wrong schedules because the schedule was set initial by UI or it was changed. Also tags can be deleted and so on.

### Describe the proposed behavior

There should be a `--overwrite=False` flag which then just skips the deployment if it already exists.

### Example Use

We like to deploy our flows via CI but won't make a deployment everytime we need to adjust the schedule. This adds additonal overhead (because of our deployment process and guidelines) and may cause in wrong configuration if someone needs to change the schedule by UI and forgot to change it in the deployment specs.

**Comment by bunchesofdonald** [comment]
Hi @sti0! The ability to load an existing deployment was [recently added](https://github.com/PrefectHQ/prefect/pull/7218) to the `Deployment` class, would exposing that flag via the CLI with a `--load-existing` option handle your use case?

**Comment by madkinsz** [comment]
@bunchesofdonald note that new flag actually allows you to _disable_ loading the existing deployment — our default behavior is to load it.

**Comment by bunchesofdonald** [comment]
Oh great point @madkinsz.

**Comment by sti0** [comment]
@bunchesofdonald I don't think so. The idea behind my request is to not overwrite/execute the deployment again if it already exists.

**Comment by madkinsz** [comment]
@billpalombi this pairs well with previous discussion of "locking" deployments such that they cannot be edited in the UI. A `--no-overwrite` flag seems reasonable to include on `prefect deployment apply`.

**Comment by billpalombi** [comment]
@madkinsz To be clear, the `--no-overwrite` flag would prevent the deployment from being updated by the command if a deployment of the same flow with the same name already exists? If so, I support that.

**Comment by desertaxle** [comment]
I believe this is solved via the introduction of:
1. the ability to set multiple schedules on a deployment
2. deployment schedule slugs to avoid overriding all schedules on a deployment via `prefect deploy`.
As such I'm closing this issue, but please comment if you think it has been closed in error!
